test/integration/explorer-handler.spec.ts

```diff
@@ -296,6 +296,141 @@ testWithComponents(() => {
+
+    alchemyNftFetcher.getNFTsForOwner = jest
+      .fn()
+      .mockResolvedValue(thirdPartyWearables.map((wearable) => wearable.urn.decentraland))
+    baseWearablesFetcher.fetchOwnedElements = jest.fn().mockResolvedValue(baseWearables)
+    content.fetchEntitiesByPointers = jest.fn(async (pointers) =>
+      pointers.map((pointer) => entities.find((def) => def.id === pointer))
+    )
+    theGraph.ethereumCollectionsSubgraph.query = jest.fn().mockResolvedValue({ nfts: onChainWearables.slice(0, 5) })
+    theGraph.maticCollectionsSubgraph.query = jest.fn().mockResolvedValue({ nfts: onChainWearables.slice(5, 10) })
+    fetch.fetch = jest.fn().mockImplementation((url) => {
+      if (url.includes('test-collection')) {
+        return {
+          ok: true,
+          json: () => ({
+            entities: generateWearableEntities(thirdPartyWearables.map((wearable) => wearable.urn.decentraland))
+          })
+        }
+      } else {
+        return { ok: true, json: () => ({ entities: [] }) }
+      }
+    })
+
+    const wallet = generateRandomAddress()
+    const r = await localFetch.fetch(`/explorer/${wallet}/wearables?compact=true`)
+
+    expect(r.status).toBe(200)
+    const response = await r.json()
+
+    expect(response).toMatchObject({
+      pageNum: 1,
+      pageSize: 100,
+      totalAmount: baseWearables.length + onChainWearables.length + thirdPartyWearables.length
+    })
+
+    // Verify all elements have compact structure (no type field, simplified entity)
+    expect(response.elements).toHaveLength(6)
+    response.elements.forEach((element: any) => {
+      // Compact response should not have a type field
+      expect(element.type).toBeUndefined()
+      expect(element.entity).toBeDefined()
+      expect(element.entity.id).toBeDefined()
+      expect(element.entity.metadata).toBeDefined()
+      expect(element.entity.metadata.id).toBeDefined()
+      expect(element.entity.metadata.data).toBeDefined()
+      expect(element.entity.metadata.data.category).toBeDefined()
+      expect(element.entity.metadata.data.representations).toBeDefined()
+      expect(element.entity.metadata.thumbnail).toBeDefined()
+
+      // Rarity is only available for on-chain wearables
+      if (element.entity.id.startsWith('urn-')) {
+        // On-chain wearables should have rarity
+        expect(element.rarity).toBeDefined()
+        expect(element.rarity).toBe('unique')
+      } else {
+        // Base wearables and third-party wearables don't have rarity
+        expect(element.rarity).toBeUndefined()
+      }
+
+      // Verify representations are simplified to only include bodyShapes
+      element.entity.metadata.data.representations.forEach((representation: any) => {
+        expect(representation.bodyShapes).toBeDefined()
+        expect(representation.bodyShapes).toBeInstanceOf(Array)
+        // Should not have other properties like contents, etc.
+        expect(Object.keys(representation)).toEqual(['bodyShapes'])
+      })
+    })
```

Comment:
Is there any chance we could assert over the real elements here? Most of the assertions check that they're defined but don't change the content of it, so we can't be really sure that they're returning what's expected.